plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -245,42 +195,20 @@ compileToReadable (Program a v t) =
 compileReadableToPlc :: (Compiling m e uni fun a, b ~ Provenance a) => Program TyName Name uni fun b -> m (PLCProgram uni fun a)
 compileReadableToPlc (Program a v t) =
   let pipeline =
-        (<$ logVerbose "  !!! floatIn")
-        >=> floatIn
-        >=> through check
-        >=> (<$ logVerbose "  !!! compileNonStrictBindings")
-        >=> NonStrict.compileNonStrictBindings False
-        >=> through check
-        >=> (<$ logVerbose "  !!! thunkRecursions")
-        >=> (withBuiltinsInfo . fmap pure . flip ThunkRec.thunkRecursions)
-        -- Thunking recursions breaks global uniqueness
-        >=> PLC.rename
-        >=> through check
+        runPass floatInPasses
+        >=> runPass (NonStrict.compileNonStrictBindingsPassSC <$> view ccTypeCheckConfig <*> pure False)
```

Comment:
You mean a `Monoid`? I guess it could be, if we drop the names from `CompoundPass`. I kind of like the names, though, it's helpful to know that you're running a specific group of passes with a name...

Perhaps:
```
data Pass m ...
 = Pass (...  -> m ...) [Condition] [BiCondition]
 | Compound [Pass]
 | Named String Pass
```
?

Reply:
I committed to monoids, I think it's pretty nice actually!